src/System Application/App/AppSource Gallery/src/AppSourceProductManager.Codeunit.al

```diff
@@ -0,0 +1,604 @@
+
+using System.Environment.Configuration;
+using System.Globalization;
+using System.Azure.Identity;
+using System.Utilities;
+using System.Environment;
+using System.Azure.KeyVault;
+using System.RestClient;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
+/// </summary>
+codeunit 2515 "AppSource Product Manager" implements "IAppSource Product Manager Dependencies"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    #region Dependency Interface implementation
+    procedure AzureADTenant_GetCountryLetterCode(): Text[2]
+    var
+        entraTenant: Codeunit "Azure AD Tenant";
+    begin
+        exit(entraTenant.GetCountryLetterCode());
+    end;
+
+    procedure AzureAdTenant_GetPreferredLanguage(): Text
+    var
+        entraTenant: Codeunit "Azure AD Tenant";
+    begin
+        exit(entraTenant.GetPreferredLanguage());
+    end;
+
+
+    procedure AzureADTenant_GetAadTenantID(): Text
+    var
+        entraTenant: Codeunit "Azure AD Tenant";
+    begin
+        exit(entraTenant.GetAadTenantID());
+    end;
+
+    procedure AzureKeyVault_GetAzureKeyVaultSecret(SecretName: Text; var Secret: SecretText);
+    var
+        KeyVault: Codeunit "Azure Key Vault";
+    begin
+        KeyVault.GetAzureKeyVaultSecret(SecretName, Secret);
+    end;
+
+    procedure EnvironmentInformation_GetApplicationFamily(): Text
+    var
+        EnvironmentInformation: Codeunit "Environment Information";
+    begin
+        exit(EnvironmentInformation.GetApplicationFamily());
+    end;
+
+    procedure EnvironmentInformation_IsSaas(): boolean
+    var
+        EnvironmentInformation: Codeunit "Environment Information";
+    begin
+        exit(EnvironmentInformation.IsSaas());
+    end;
+
+    procedure Language_GetFormatRegionOrDefault(FormatRegion: Text[80]): Text
+    var
+        Language: Codeunit Language;
+    begin
+        exit(Language.GetFormatRegionOrDefault(FormatRegion));
+    end;
+
+    procedure RestClient_GetAsJSon(var RestClient: Codeunit "Rest Client"; RequestUri: Text): JsonToken
+    begin
+        exit(RestClient.GetAsJSon(RequestUri));
+    end;
+
+    procedure UserSettings_GetUserSettings(UserSecurityId: Guid; var TempUserSettingsRecord: record "User Settings" temporary)
+    var
+        UserSettings: Codeunit "User Settings";
+    begin
+        UserSettings.GetUserSettings(Database.UserSecurityID(), TempUserSettingsRecord);
+    end;
+
+    #endregion
+    procedure SetDependencies(SpecificDependencies: Interface "IAppSource Product Manager Dependencies")
+    begin
+        Dependencies := SpecificDependencies;
+        IsDependenciesSet := true;
+    end;
+
+    #region Product helpers 
+    /// <summary>
+    /// Opens Microsoft AppSource web page.
+    /// </summary>
+    procedure OpenAppSource()
+    begin
+        Init();
+        Hyperlink(StrSubstNo(AppSourceUriLbl, Dependencies.Language_GetFormatRegionOrDefault('en-us')));
```

Comment:
shouldn't we get the format and region from user settings instead of always en-us? There's even the user settings action which says "ToolTip = 'Locale will be used to determine the market and language will be used to determine the language of the app details listed here.';"

Reply:
To address in this PR